ui/app/components/secret-engine/configure-aws.ts

```diff
@@ -72,40 +102,78 @@ export default class ConfigureAwsComponent extends Component<Args> {
     this.disableAccessType = wifAttributesSet || iamAttributesSet;
   }
 
-  @task
-  @waitFor
-  *save(event: Event): Generator<Promise<boolean | LeaseConfigModel | RootConfigModel>> {
-    event.preventDefault();
-    this.resetErrors();
-    const { leaseConfig, rootConfig } = this.args;
-    // Note: aws/root-config model does not have any validations
-    const isValid = this.validate(leaseConfig);
-    if (!isValid) return;
-    // Check if any of the models' attributes have changed.
-    // If no changes to either model, transition and notify user.
-    // If changes to either model, save the model(s) that changed and notify user.
-    // Note: "backend" dirties model state so explicity ignore it here.
-
-    const leaseAttrChanged =
-      Object.keys(leaseConfig.changedAttributes()).filter((item) => item !== 'backend').length > 0;
-    const rootAttrChanged =
-      Object.keys(rootConfig.changedAttributes()).filter((item) => item !== 'backend').length > 0;
-
-    if (!leaseAttrChanged && !rootAttrChanged) {
-      this.flashMessages.info('No changes detected.');
-      this.transition();
-    }
+  save = task(
+    waitFor(async (event: Event) => {
+      event.preventDefault();
+      this.resetErrors();
+      const { leaseConfig, rootConfig } = this.args;
+      // Note: aws/root-config model does not have any validations
+      const isValid = this.validate(leaseConfig);
+      if (!isValid) return;
+
+      // do issuer save and side effects if enterprise and isser saved
+      if (this.version.isEnterprise && this.issuerConfig.dirty) {
+        await this.updateIssuer.perform(this.issuerConfig);
+      }
+
+      // Check if any of the models' attributes have changed.
+      // If no changes to either model, transition and notify user.
+      // If changes to either model, save the model(s) that changed and notify user.
+      // Note: "backend" dirties model state so explicity ignore it here.
 
-    const rootSaved = rootAttrChanged ? yield this.saveRoot() : false;
-    const leaseSaved = leaseAttrChanged ? yield this.saveLease() : false;
+      const leaseAttrChanged =
+        Object.keys(leaseConfig.changedAttributes()).filter((item) => item !== 'backend').length > 0;
```

Comment:
FWIW I like using `some()` here so it returns a boolean and then we don't have to also check for `length > 0`